sdk/core/core-http/test/policies/atomSerializationPolicyTests.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { assert } from "chai";
+import { HttpHeaders } from "../../lib/httpHeaders";
+import { HttpOperationResponse } from "../../lib/httpOperationResponse";
+import { HttpClient, AtomXmlOperationSpec, AtomXmlSerializer } from "../../lib/coreHttp";
+import { atomSerializationPolicy } from "../../lib/policies/atomSerializationPolicy";
+import { RequestPolicyOptions } from "../../lib/policies/requestPolicy";
+import { WebResource } from "../../lib/webResource";
+
+describe("atomSerializationPolicy", function() {
+  it("should throw an error if receiving a non-XML response body", async function() {
+    const request: WebResource = createRequest({
+      serializer: new MockSerializer()
+    });
+    const mockClient: HttpClient = {
+      sendRequest: (req) =>
+        Promise.resolve({
+          request: req,
+          status: 200,
+          headers: new HttpHeaders({ "Content-Type": "application/json" }),
+          bodyAsText: `{ "simple": "JSONobject" }`
+        })
+    };
+
+    const policy = atomSerializationPolicy().create(mockClient, new RequestPolicyOptions());
+    try {
+      await policy.sendRequest(request);
+      assert.deepEqual(true, false, "Error must be thrown");
+    } catch (err) {
+      assert.deepEqual(err.message, "ResponseNotInAtomXMLFormat");
+      assert.deepEqual(err.code, "PARSE_ERROR");
+    }
+  });
+
+  it("with xml response body, application/xml content-type and AtomXMLOperationSpec", async function() {
+    const request: WebResource = createRequest({
+      serializer: new MockSerializer()
+    });
+
+    const expectedResult = {
+      LockDuration: "PT2M",
+      MaxSizeInMegabytes: "1024",
+      _: {
+        ContentRootElement: "QueueDescription",
+        id:
+          "https://servicebuslocalperftestspremium1.servicebus.windows.net/testQueuePath4?api-version=2017-04",
```

Comment:
In case this is pointing to a real resource, please update to use dummy values since we are anyway using a mock client

Reply:
Resolving as tests have been updated